packages/devtools_app/lib/src/flutter/screen.dart

```diff
@@ -76,6 +76,28 @@ abstract class Screen {
   }
 }
 
+/// Defines a page shown in the DevTools TabBar if [conditionalLibrary] is
+/// present in the connected application.
+abstract class ConditionalScreen extends Screen {
```

Comment:
It is worth adding a bit more of a comment for this class as it will have customers who aren't familiar with our code.
Maybe given an example showing how to use it for `package:provider`?

Also indicate limitations such as:
These screens will be completely hidden if the screen isn't available unlike pages such as the inspector that will intentionally show up as disabled if they are not available.

Reply:
added docs